lib/submit-analytics.js

```diff
@@ -33,50 +104,184 @@ function sha256(data) {
     return hash.digest('hex');
 }
 
-if (isAnalyticsDisabled()) {
-    module.exports = function(){};
-} else {
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {[webhookProdDetails|webhookStageDetails|mixpanelDetails]} requestFactory Submission URL generator
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(requestFactory, payload) {
+    const https = require('https');
+
+    return new Promise((resolve, reject) => {
+        const webhookRequest = new requestFactory().buildRequest(payload);
+        var req = https.get(webhookRequest, function (result) {
+            result.on('data', (data) => {
+                // we don't log the data that is returned -- we're just interested in the header
+            });
+        });
+
+        req.on('response', response => {
+            response.on('end', () => {
+                resolve(
+                    {
+                        statusCode:  response.statusCode,
+                        statusMessage: response.statusMessage,
+                        errorMessage: '',
+                    });
+                return;
+            });
+
+            response.on('error', error => {
+                reject(
+                    {
+                        statusCode:  response.statusCode,
+                        statusMessage: response.statusMessage,
+                        errorMessage: error,
+                    });
+                return;
```

Comment:
No need to return here - blocks automatically and implicitly `return undefined`.
```suggestion
                return;
```